opm/simulators/wells/WellGroupHelpers.hpp

```diff
@@ -422,11 +471,12 @@ namespace Opm {
                     continue;
             }
 
-            if ( (groupTmp.isProductionGroup() && !isInjector) ||
-                 (groupTmp.isInjectionGroup() && isInjector) )
-            {
+            if ( (groupTmp.isProductionGroup() && !isInjector)) {
                 groupTotalGuideRate += guideRate->get(groupName, groupTarget);
+            } else if ((groupTmp.isInjectionGroup() && isInjector)){
+                assert(false);
```

Comment:
I do not understand why this can never happen.

Reply:
The guideRate object is only for injectors. I can change to a throw.